contrib/babelfishpg_tsql/src/pl_funcs-2.c

```diff
@@ -350,6 +351,60 @@ coalesce_typmod_hook_impl(const CoalesceExpr *cexpr)
 	return exprTypmod((Node *) linitial(cexpr->args));
 }
 
+void
+check_restricted_stored_procedure(Oid proc_id)
+{
+	HeapTuple		proctup;
+	Form_pg_proc		procform;
+	const char		*procname;
+	Oid		schema_oid = InvalidOid;
+	Oid		dbo_oid = InvalidOid;
+	bool		is_restricted = false;
+
+	/*
+	 * List of procedure names that are not allowed to be dropped. These procedures 
+	 * are considered essential or restricted due to security or operational reasons.
+	 */
+	static const char	*restricted_procedures[] = {
```

Comment:
Do we need to make these variables static if these are only needed for the context of this one specific function?

Reply:
By defining it as a static, we would saving it from reinitialising it from every call to this function. Initialisation scope would be til the end of backend but variable scope would be local within the function it defined.